packages/shared_preferences/shared_preferences_platform_interface/README.md

```diff
@@ -1,17 +1,26 @@
 # shared_preferences_platform_interface
 
-A common platform interface for the [`shared_preferences`][1] plugin.
+A common platform interface for the [`shared_preferences`][1] plugins.
 
 This interface allows platform-specific implementations of the `shared_preferences`
 plugin, as well as the plugin itself, to ensure they are supporting the
 same interface.
 
 # Usage
 
-To implement a new platform-specific implementation of `shared_preferences`, extend
-[`SharedPreferencesPlatform`][2] with an implementation that performs the
-platform-specific behavior, and when you register your plugin, set the default
-`SharedPreferencesLoader` by calling the `SharedPreferencesPlatform.loader` setter.
+To implement a new platform-specific implementation of `shared_preferences`, 
```

Comment:
I don't think we want to describe this is something where people should make an individual decision by file; instead I would say:

```
To implement a new platform-specific implementation of `shared_preferences`, extend
[`SharedPreferencesPlatform`][2] and [`SharedPreferencesAsyncPlatform`][3] with
implementations that perform the platform-specific behaviors, and when you register
your plugin, set the default `SharedPreferencesStorePlatform` and
`SharedPreferencesAsyncPlatform` by calling the `SharedPreferencesPlatform.instance`
and `SharedPreferencesAsyncPlatform.instance` setters.

Please note that the plugin tooling only registers the native and/or Dart classes
listed in your package's `pubsec.yaml`, so if you intend to implement more than
one class, you will need to manually register the second class
(as can be seen in the Android and iOS implementations).
```